HW3/P4/median_filter.cl

```diff
@@ -1,5 +1,33 @@
+// This exercise was discussed with Andrew Petschek
+
+
 #include "median9.h"
 
+// Creating a function that fills the buffer with the right values (like in PB 5)
+float
+get_buff(__global __read_only float *in_values,
+                  int w, int h,
+                  int x, int y)
+// Checking multiple cases:
+{   if (x<0){
+      if (y<0) {return in_values[0];} // top left corner of the halo
+      else if (y >= h) {return in_values[(h-1) * w];} //bottom left corner of the halo
+      else {return in_values[y*w];} //left side of the halo
+    }
+
+    else if (x>=w){
+      if (y<0) {return in_values[w-1];} //top right corner of the halo
+      else if (y>=h) {return in_values[(h-1) * w + w - 1];} //bottom right corner of the halo
+      else {return in_values[(w-1) + y*w];} // right side of the halo
+    }
+
+    else{
+      if (y<0) {return in_values[x];} //top side of the halo
+      else if (y>=h) {return in_values[x + (h-1) * w];} //lower side of the halo
+      else {return in_values[x + w*y];} // Regular case (one without any exception)
+    }
+}
+
```

Comment:
Quite cumbersome. An easier way to do it might be:
if (x < 0) x = 0;
if (x >= w) x = w-1;
if (y < 0) y = 0;
if (y >= h) y = h-1;
return in_values[x + w*y].